app/src/handler.rs

```diff
@@ -120,22 +123,33 @@ pub fn handle_apdu(
     Ok(())
 }
 
+const MAX_TOKEN_SIZE: u8 = 5;
+
 fn handle_sign_tx(
     apdu_header: &ApduHeader,
     data: &[u8],
     sign_tx_context: &mut SignTxContext,
     tx_reviewer: &mut TxReviewer,
 ) -> Result<(), ErrorCode> {
     match apdu_header.p1 {
-        0 if data.len() < 23 => Err(ErrorCode::BadLen),
+        0 if data.len() < 21 => Err(ErrorCode::BadLen), // 20 bytes path + 1 byte token size
```

Comment:
We need one more byte for version: `20 bytes path + 1 byte version (only 0 for now) + 1 byte token size`

Just to be forward-compatible as we might support NFT in the future.